### ISSUE TITLE
WeaponModModule: Global Values Still Absolutely F*cking Cursed

### ISSUE THREAD

**Issue by the-almighty-carl** [simon-wh/PAYDAY-2-BeardLib#234]
Still continuing the mess from #211 here. Woo.

The fix for #211 pressed about five days ago did in fact fix the problem of weapon mods with custom global values becoming duplicated - having a weapon mod with a custom global value will no longer duplicate said weapon mod. The problem, however, is that _**giving even a single weapon mod a global value almost completely breaks the game now.**_

Let me take a step back and explain what I mean. Having no custom global values on any weapon mods, being the normal case, led to normal behavior in the game. However, the moment I gave a custom global value to one of my custom weapon mods, all hell broke loose. The uses_parts tables for weapons melted into each other (example: parts from Pawcio's Walther P99 were suddenly available on the base-game Chimano 88), and more alarmingly, add-on weapon mods seemed to have their unit paths broken at random (example: custom weapons that could be previewed before would now cause the trademark infinite hang of a broken unit path when hitting the "preview weapon" button). Note that it *probably* isn't actual destruction of unit paths at random, and is more likely the game being stubborn about something related to global values again, but still worrying.

I strongly suspect that some oversight in the new function overwriting `BlackMarketManager:get_dropable_mods_by_weapon_id` is causing this behavior. I have absolutely no idea what the hell any of this is for certain, but I'd put my money on a `break` being missing somewhere, thus making get_dropable_mods_by_weapon_id insert every single weapon mod with drop set to `true` into the uses_parts table of every single weapon in the game (or some other equally dumb thing that causes this end result).

tldr oof

**Comment by Luffyyy** [comment]
Are you sure you are using the latest version of the mod as there were unrelated changes

**Comment by Luffyyy** [comment]
And no the function edit just removes duplicates

**Comment by the-almighty-carl** [comment]
Was on the latest version of everything I had installed at the time of writing and still am.

So... I guess I can go die, then.

**Comment by Luffyyy** [comment]
What? So you weren't on the latest  GitHub commit?

**Comment by Luffyyy** [comment]
c a r l

**Comment by the-almighty-carl** [comment]
That... That's what that implies, yes.

**Comment by the-almighty-carl** [comment]
no
